maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -1571,13 +1566,15 @@ private void convertGroupParameters()
 
     private void convertJunitEngineParameters()
     {
-        if ( isNotBlank( getJunitIncludeEngine() ) )
+        if ( getIncludeJUnit5Engines() != null )
         {
-            getProperties().setProperty( JUNIT_INCLUDE_ENGINE_PROP, getJunitIncludeEngine() );
+            String includeJUnit5Engines = Arrays.toString( getIncludeJUnit5Engines() ).replaceAll( "\\[\\]", "" );
```

Comment:
Pls do not use `Arrays.toString`. Nobody does this way. Use a typical loop. In Java 8 we will have string joinrer but now pls use a loop and StringBuilder.